### PR TITLE
Ros2ProjectTemplate fixes to monolithic build

### PR DESCRIPTION
**Pull request by lewandowskikacper**
This applies suggestion from https://github.com/o3de/o3de-extras/issues/493.

~~There are linker issues still present after those changes.~~

The linker issues were caused by dependency on AWSCore Gem, that is now removed.

### CODE REVIEW

File: Gems/ROS2/Code/Source/FactorySimulation/ConveyorBeltComponent.cpp [hunk 1]
```diff
@@ -61,7 +61,7 @@ namespace ROS2
 
     void ConveyorBeltComponent::Activate()
     {
-        AzPhysics::SystemInterface* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get();
+        [[maybe_unused]] AzPhysics::SystemInterface* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get();
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/FactorySimulation/ConveyorBeltComponent.cpp:64]: in each case, we don't actually need to run it in release builds. Please move to asserts, as in:
` AZ_Assert(AZ::Interface<AzPhysics::SystemInterface>::Get(), "No physics system");`
Apply to other cases.

File: Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp [hunk 2]
```diff
@@ -125,7 +125,7 @@ namespace ROS2
                 auto* prismaticComponent =
                     azrtti_cast<PhysX::JointComponent*>(Utils::GetGameOrEditorComponent<PhysX::PrismaticJointComponent>(entity));
                 auto* articulationComponent = Utils::GetGameOrEditorComponent<PhysX::ArticulationLinkComponent>(entity);
-                bool classicJoint = hingeComponent || prismaticComponent;
+                [[maybe_unused]] bool classicJoint = hingeComponent || prismaticComponent;
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp:128]: this one can stay as is - good suggestion

**Review by adamdbrw — COMMENTED** [review]
Does this PR introduce new issues, or does it simply not resolve all issues present?

**Comment by lewandowskikacper** [comment]
> Does this PR introduce new issues, or does it simply not resolve all issues present?

It solves issues with this Template now.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Comment by lewandowskikacper** [comment]
I also checked the [Ros2FleetRobotTemplate](https://github.com/o3de/o3de-extras/tree/development/Templates/Ros2FleetRobotTemplate), [Ros2RoboticManipulationTemplate](https://github.com/o3de/o3de-extras/tree/development/Templates/Ros2RoboticManipulationTemplate). They both compile in profile config.

In release config, compilation fails due to "unused but set variable" caused by `result` in https://github.com/o3de/o3de/blob/60a1e5a368282a13654f45d8c1aa7be2192dfcaa/Gems/Atom/RPI/Code/Source/RPI.Public/RPISystem.cpp#L299C1-L306C21. If that is fixed, it succeeds.

**Review by michalpelka — APPROVED** [review]

**Pull request merged**